c/logging.c

```diff
@@ -685,14 +799,41 @@ void zowelog(LoggingContext *context, uint64 compID, int level, char *formatStri
       lastResortLog(message);
       return;
     }
+
+    ACEE *acee;
+    acee = getAddressSpaceAcee();
+    char user[7] = { 0 }; // wil this always be 7?
+    snprintf(user,7,acee->aceeuser+1);
+
+    char *locationInfo; // largest possible variation in size
+    getLocationData(path,line,&locationInfo,compID,component); // location info is allocated in getLocationData
+
+    char logLevel[8] = { 0 }; // logLevel will be a max of 8 characters
+    getLevel(level, logLevel);
+
+    char timestamp[24] = { 0 }; // formatted date is 24 characters
+    getLogTimestamp(timestamp); // UTC time
+
+    pthread_t threadID = pthread_self();
```

Comment:
It's my understanding that zowelog is used by both LE and Metal/C runtime environments. pthread_self() is not valid for Metal/C so you will need to handle both LE and Metal/C (using the current TCB address).